pkg/crypto/crypto.go

```diff
@@ -758,15 +758,18 @@ func GetServerCert(certFile, keyFile string, hostnames sets.String) (*TLSCertifi
 	}
 
 	cert := server.Certs[0]
-	ips, dns := IPAddressesDNSNames(hostnames.List())
-	missingIps := ipsNotInSlice(ips, cert.IPAddresses)
-	missingDns := stringsNotInSlice(dns, cert.DNSNames)
-	if len(missingIps) == 0 && len(missingDns) == 0 {
+	stringCertIPs := make([]string, len(cert.IPAddresses))
+	for i, ip := range cert.IPAddresses {
+		stringCertIPs[i] = ip.String()
+	}
+	certNames := sets.NewString(stringCertIPs...)
```

Comment:
you can create the set and just insert to it to save the slice allocation

Reply:
also, I believe `sets.NewString` is deprecated, use `sets.New[string]()`

Reply:
Good point on the first one, will do.
About the deprecation, it is indeed, but it implies changing the public API for the library (because of the comparison with the `hostnames` set. Unless I copy contents into a new `sets.New[string]()`)

Reply:
First comment done.

Reply:
I see, you can leave the 2nd part as is for now, then.